cmd/juju/controller/showcontroller.go

```diff
@@ -139,6 +139,15 @@ func (c *showControllerCommand) Run(ctx *cmd.Context) error {
 		}
 
 		var details ShowControllerDetails
+
+		// If user lacks sufficient access level to display controller details,
+		// append the empty controller details and attach a permission error.
+		if !permission.Access(access).EqualOrGreaterControllerAccessThan(permission.SuperuserAccess) {
```

Comment:
I have no idea myself, but should it always be compared to a `superuser`, or should this be a case of does the user have access to controller through access controls.

i.e. `juju grant $USER read controller`